src/ansys/fluent/core/services/field_data.py

```diff
@@ -1221,12 +1234,56 @@ def get_scalar_field_data(
                 for surface_id in surface_ids
             }
 
+    @deprecate_argument(
+        old_arg="surface_name",
+        new_arg="surface_names",
+        converter=lambda old_arg_val: [old_arg_val] if old_arg_val else None,
+        deprecation_class=PyFluentDeprecationWarning,
+    )
+    @deprecate_argument(
+        old_arg="data_type",
+        new_arg="provide_vertices",
+        converter=lambda old_arg_val: (
+            True if old_arg_val == SurfaceDataType.Vertices else False
+        ),
+        deprecation_class=PyFluentDeprecationWarning,
+        is_last_instance_of_old_arg=False,
+    )
+    @deprecate_argument(
+        old_arg="data_type",
+        new_arg="provide_faces",
+        converter=lambda old_arg_val: (
+            True if old_arg_val == SurfaceDataType.FacesConnectivity else False
+        ),
+        deprecation_class=PyFluentDeprecationWarning,
+        is_last_instance_of_old_arg=False,
+    )
+    @deprecate_argument(
+        old_arg="data_type",
+        new_arg="provide_faces_centroid",
+        converter=lambda old_arg_val: (
+            True if old_arg_val == SurfaceDataType.FacesCentroid else False
+        ),
+        deprecation_class=PyFluentDeprecationWarning,
+        is_last_instance_of_old_arg=False,
+    )
+    @deprecate_argument(
+        old_arg="data_type",
+        new_arg="provide_faces_normal",
+        converter=lambda old_arg_val: (
+            True if old_arg_val == SurfaceDataType.FacesNormal else False
+        ),
+        deprecation_class=PyFluentDeprecationWarning,
+    )
     def get_surface_data(
         self,
-        data_type: SurfaceDataType,
```

Comment:
IMO the older interface looks better. Why do we need to change it?

Reply:
Note that, there is a [Flag](https://docs.python.org/3/library/enum.html#enum.Flag) class to compose Enum values if that is what we are looking for here.

Reply:
If this interface is better, then we need to have "data_types" here as well as in add_suface_request. I'll  do that change then.
Thank you.

Reply:
I'll discuss with you this point.